openedx/core/djangoapps/programs/utils.py

```diff
@@ -150,8 +197,14 @@ def progress(self, programs=None, count_only=True):
             for course in program['courses']:
                 if self._is_course_complete(course):
                     completed.append(course)
-                elif self._is_course_in_progress(course):
-                    in_progress.append(course)
+                elif self._is_course_enrolled(course):
+                    course_in_progress = self._is_course_in_progress(now, course)
+                    course_copy = deepcopy(course)
+                    if course_in_progress:
+                        in_progress.append(course_copy)
+                    else:
+                        course_copy['expired'] = not course_in_progress
+                        not_started.append(course_copy)
                 else:
                     not_started.append(course)
```

Comment:
i thought you just wanted the things in the if block to be a copy, perhaps there are other approaches that you would be happy with?  
perhaps we can just deepcopy the program once?